src/porepy/numerics/ad/operators.py

```diff
@@ -108,6 +166,28 @@ def __init__(
                 " interfaces, subdomains or boundary grids."
             )
 
+        self.func: Callable[..., float | np.ndarray | AdArray]
```

Comment:
I introduced the declaration of this attribute to handle the abstraction in `AbstractFunction`.
This will also come in handy in the future, in case we optimize the evaluation of the operator tree towards a more efficient, functional approach, (E.g., any operation in the AD operator sense can be represented as a function, which is a more general format)